index.js

```diff
@@ -63,15 +59,14 @@ const walkMutate = (input, predicate, mutator) => {
         input[key] = walkMutate(input[key], predicate, mutator);
       }
     }
-    return input;
   }
   return input;
 };
 
-const normalizeLink = (responseClone, link, removeUnresolved) => {
-  const resolvedLink = getLink(responseClone, link);
+const normalizeLink = (allEntries, link, removeUnresolved) => {
+  const resolvedLink = getLink(allEntries, link);
   if (resolvedLink === undefined) {
-    return removeUnresolved ? undefined : link;
+    return removeUnresolved && isLink(link) ? undefined : link;
```

Comment:
`isLink` is the constraint of `walkMutate`, so `normalizeLink` gets called **only** for links, no need to check again

Reply:
Ooops, that one was a test and went in on accident. Good catch